### DISCUSSION TITLE
Atom family setter doesn't seem to update value

### DISCUSSION THREAD

**Discussion by blissdev** [pmndrs/jotai#1949] [Q&A] [answered]
I think I'm likely using it wrong, but could someone point me in the right direction?

I setup this example: https://codesandbox.io/s/modest-farrell-yn9bq8

The map values never seem to get set, and I'm not sure why.

Overall, I'd like to be able to set map values within the react lifecycle, as in my real usecase it will be a grid component where grid cells load their data asyncronously when they come into view.

Follow-up to question here: https://github.com/pmndrs/jotai/discussions/1942#discussioncomment-5958953

**Comment by dai-shi** [comment, marked as answer]
Oops, I overlooked the code snippet in https://github.com/pmndrs/jotai/discussions/1942#discussioncomment-5958953.

Here's the fix: https://codesandbox.io/s/pensive-wiles-j6wy1e?file=/src/App.tsx

**Comment by blissdev** [comment]
@dai-shi Thanks so much for the quick response!

It looks like the main difference is calling `set()`? Should the second example in the docs be updated to use this as well? Or is that not the intent there?

Lastly, do you think this the appropriate solution for my case? Or would atom of atoms be better?

**Reply by dai-shi** [reply]
Yes, the main issue is calling `set()`.

It's appropriate for your case. Only a caveat is that `atomFamily` is a `Map`, so unless you remove explicitly, all atom configs are kept in memory, and so are atom values.

> Should the second example in the docs be updated to use this as well? Or is that not the intent there?

Which one is it in the docs?

**Reply by blissdev** [reply]
Second under examples:

```js
import { atom } from 'jotai'
import { atomFamily } from 'jotai/utils'

const todoFamily = atomFamily((name) =>
  atom(
    (get) => get(todosAtom)[name],
    (get, set, arg) => {
      const prev = get(todosAtom)
      return { ...prev, [name]: { ...prev[name], ...arg } }
    }
  )
)
```

**Reply by dai-shi** [reply]
Hmmm, I think it's a mistake from the old version (maybe even before v1).
Can you send a PR to fix it please?

**Reply by blissdev** [reply]
Will do.
